### PR TITLE
add delete action (#26)

### LINKED ISSUE #26: Delete service/revision action

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #26 
Classic delete action for services/revision with multiple selection available

### CODE REVIEW

**Review by evidolob — APPROVED** [review]
Maybe we can add keybinding for delete action?

**Comment by lstocchi** [comment]
> Maybe we can add keybinding for delete action?

This could be a good idea that can be extended to other actions. We could add some setting/customization where the user can set the keybinding for different actions. Opening a new issue so we can talk about it. :+1:

File: src/main/java/com/redhat/devtools/intellij/knative/actions/DeleteAction.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+public class DeleteAction extends KnAction {
+    public DeleteAction() {
+        super(true, KnServiceNode.class, KnRevisionNode.class);
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath[] path, Object[] selected, Kn kncli) {
+        ParentableNode[] elements = Arrays.stream(selected).map(item -> getElement(item)).toArray(ParentableNode[]::new);
+        DeleteDialog deleteDialog = UIHelper.executeInUI(() -> {
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/knative/actions/DeleteAction.java:40]: I don't get it, actions are executed in UI thread by default, seems useless for me

**Review by jeffmaury — APPROVED** [review]
LGTM. However, I have the following comment: I deleted first the single revision for the service, got no refresh, and after I refreshed, both service and revision were displayed in red color, not sure this is the expected behavior.
Deleting the service got the tree refreshed correctly

**Comment by lstocchi** [comment]
@jeffmaury i am not able to replicate it. By deleting a revision from a service works fine for me (gif below). 
Can you copy/paste your kservice? When you delete the only existing revision, knative should recreate a new one but if you see it in red after the refresh it's like it failed somehow.

delete-revision

**Pull request merged**